src/axl.c

```diff
@@ -280,11 +335,31 @@ int AXL_Create (axl_xfer_t xtype, const char* name)
      *       SOURCE
      *     STATE
      *       CREATED */
-    kvtree* file_list = kvtree_set_kv_int(axl_file_lists, AXL_KEY_HANDLE_UID, id);
-    kvtree_util_set_str(file_list, AXL_KEY_UNAME, name);
-    kvtree_util_set_int(file_list, AXL_KEY_XFER_TYPE, xtype);
-    kvtree_util_set_int(file_list, AXL_KEY_STATUS, AXL_STATUS_SOURCE);
-    kvtree_util_set_int(file_list, AXL_KEY_STATE, (int)AXL_XFER_STATE_CREATED);
+    if (reload_from_state_file) {
+        if (axl_get_info(id, &file_list, &old_xtype, &old_xstate) != AXL_SUCCESS) {
+            AXL_ERR("Couldn't get kvtree info\n");
+            return AXL_FAILURE;
+        }
+
+        if (xtype != old_xtype) {
```

Comment:
Let's also check that the name matches.

I haven't fully thought through things yet, but I feel like we'll have a race condition we need to figure out here when we get to that.

Reply:
I was thinking they may want to set a new name, like if they include the job ID in the name ("simulation42").

Reply:
Ah, I see.  In that case, I think we'd need a line here to overwrite the existing name value.
```
/* Allow caller to assign a new name */
kvtree_util_set_str(file_list, AXL_KEY_UNAME, name);
```

Reply:
Did you update the code for this in your push?  I think it's showing as the same for me.

Reply:
@tonyhutter , this comment came in as part of my last review, but since you didn't change it or respond in your last push, I'm guessing it got overlooked.

Reply:
Thanks, I did overlook that.  It's fixed in the push I just did:
```C
diff --git a/src/axl.c b/src/axl.c
index 68680ff..703e8df 100644
--- a/src/axl.c
+++ b/src/axl.c
@@ -358,7 +358,7 @@ int __AXL_Create(axl_xfer_t xtype, const char* name, const char *state_file)
             xtype = axl_detect_native_xfer();
         }
 
-        kvtree* file_list = axl_kvtrees[id];
+        file_list = axl_kvtrees[id];
         kvtree_util_set_int(file_list, AXL_KEY_XFER_TYPE, xtype);
         kvtree_util_set_int(file_list, AXL_KEY_STATUS, AXL_STATUS_SOURCE);
         kvtree_util_set_int(file_list, AXL_KEY_STATE, (int)AXL_XFER_STATE_CREATED);
```